### DISCUSSION TITLE
Morphia query should also return Documents /  json and not just mapped objects.

### DISCUSSION THREAD

**Discussion by SOCRATESZZ** [MorphiaOrg/morphia#3156] [General]
Example: 
I have an Book entity. it has 
class Book {
String id;
String author;
int price;
Date manufacturingDate; 
. 
.
. 
}

when I am trying to query using Projections.include(id, author). It gives me entire Book object with id and author intitialized and other fields set to null. 
(Actual Problem)

1. this is acceptable  when we know the entities we are dealing with. But if I had to make it dynamic we dont know which all fields did we fetched with.  
2. It increases the over head of handling objects and accessing data. Which can solved by returning Document / Json.

**Comment by evanchooly** [comment]
The `Query` object is supposed to be typed so this is behaving as expected.  If you'd like to project to a different return type, you can do that with an `Aggregation` and pass in the desired type in the [`execute()`](https://morphia.dev/morphia/2.4/aggregations.html#_aggregations) call.
